api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -94,15 +94,15 @@ public void setPatientId(Integer patientId) {
 	}
 	
 	public Date getStartDate() {
-		return startDate;
+		return (Date) startDate.clone();
 	}
 	
 	public void setStartDate(Date startDate) {
-		this.startDate = startDate;
+		 this.startDate = startDate != null ? new Date(startDate.getTime()) : null;
```

Comment:
What is the difference between this and `this.startDate = new Date(startDate.getTime())` ?

Reply:
hey @f4ww4z. The current code addresses the possibility of null values too.